### PR TITLE
updates object_detection/eval.py to be python3 compatible

### PR DESCRIPTION
**Pull request by bmabey**
With these changes the new `object_detection/eval.py` script runs under python 3.

BTW, the `train.py` script has python 3 compat issues as well but they seem to be problems in tensorflow itself when I have opened [another PR](https://github.com/tensorflow/tensorflow/pull/11039) for.

### CODE REVIEW

File: object_detection/utils/metrics.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from __future__ import division
 
 import numpy as np
+from six.moves import range
```
Comment (derekjchow) [inline object_detection/utils/metrics.py]: Could you change this to `from six import moves` and use moves.range(...) below? This avoids redefining the built-in range function.

File: object_detection/utils/np_box_list.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 """Numpy BoxList classes and functions."""
 
 import numpy as np
+from six.moves import range
```
Comment (derekjchow) [inline object_detection/utils/np_box_list.py]: Ditto.

**Comment by bmabey** [comment]
@derekjchow Okay, I made the requested changes and amended my commit with them.

**Comment by derekjchow** [comment]
Look great! Thanks again for the change Ben!

**Pull request merged**